code/datums/keybinding/living.dm

```diff
@@ -82,3 +82,58 @@
 	var/mob/living/L = user.mob
 	L.look_reset()
 	return TRUE
+
+
+/datum/keybinding/living/select_intent
+	/// The intent this keybinding will switch to.
+	var/intent
+
+/datum/keybinding/living/select_intent/can_use(client/user)
+	. = ..()
+	if(!.)
+		return
+	return !iscyborg(user.mob) && (intent in user.mob?.possible_a_intents) // The cyborg check is because cyborgs have their own swap intent hotkey, and we don't want to mess that up.
+
+/datum/keybinding/living/select_intent/down(client/user)
+	. = ..()
+	if(.)
+		return
+	user.mob?.a_intent_change(intent)
+	to_chat(user, "<span class='notice'>Intent set to <b>[intent]</b>.</span>")
```

Comment:
Humans have a HUD which makes this really spammy and redundant. Make this isolated to nonhumans or things without the intent HUD.

Reply:
Or preferably add the intent hud to everything since everything now has intents.